### PR TITLE
Reduce match expressions

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

Reduces some unnecessary match expressions, that made the code a bit more verbose than it needs to be, like matching against a `bool` or against an `Option`. In another instance one could one-line things by iterating over the associated `Option`.

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

- I ran the unit-tests, and all succeeded.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bee-common/bee-common/src/packable.rs [hunk 1]
```diff
@@ -92,22 +92,15 @@ impl<P: Packable> Packable for Option<P> {
     type Error = OptionError<P::Error>;
 
     fn packed_len(&self) -> usize {
-        true.packed_len()
-            + match self {
-                Some(p) => p.packed_len(),
-                None => 0,
-            }
+        true.packed_len() + self.iter().map(|p| p.packed_len()).sum::<usize>()
```
Comment (thibault-martinez) [inline bee-common/bee-common/src/packable.rs]: You can probably `map(Packable::packed_len)` to avoid the closure

**Review by thibault-martinez — DISMISSED** [review]

File: bee-common/bee-common/src/packable.rs [hunk 2]
```diff
@@ -57,7 +57,7 @@ where
     type Error = P::Error;
 
     fn packed_len(&self) -> usize {
-        0u64.packed_len() + self.iter().map(|x| x.packed_len()).sum::<usize>()
+        0u64.packed_len() + self.iter().map(Packable::packed_len).sum::<usize>()
```
Comment (Adam-Gleave) [inline bee-common/bee-common/src/packable.rs:60]: Nice, I'll have to remember that.

**Review by Adam-Gleave — DISMISSED** [review]

File: bee-common/bee-common/src/packable.rs [hunk 3]
```diff
@@ -92,22 +92,15 @@ impl<P: Packable> Packable for Option<P> {
     type Error = OptionError<P::Error>;
 
     fn packed_len(&self) -> usize {
-        true.packed_len()
-            + match self {
-                Some(p) => p.packed_len(),
-                None => 0,
-            }
+        true.packed_len() + self.iter().map(Packable::packed_len).sum::<usize>()
```
Comment (zesterer) [inline bee-common/bee-common/src/packable.rs]: `self.as_ref().map_or(0, P::packed_len)`

is shorter than

`self.iter().map(Packable::packed_len).sum::<usize>()`

;)

(I assume we're code-golfing here?)

Comment (thibault-martinez) [reply bee-common/bee-common/src/packable.rs]: I just saw I was using the first somewhere else and was about to comment the same!

**Review by zesterer — DISMISSED** [review]

**Pull request merged**